drivers/spi/spi_gd32.c

```diff
@@ -221,13 +225,23 @@ static int spi_gd32_transceive(const struct device *dev,
 
 	spi_context_cs_control(&data->ctx, true);
 
+#ifdef CONFIG_SPI_GD32_INTERRUPT
+	SPI_STAT(cfg->reg) &= ~(SPI_STAT_RBNE | SPI_STAT_TBE | SPI_GD32_ERR_MASK);
+	SPI_CTL1(cfg->reg) |= (SPI_CTL1_RBNEIE | SPI_CTL1_TBEIE | SPI_CTL1_ERRIE);
+	ret = spi_context_wait_for_completion(&data->ctx);
+#else
 	do {
 		ret = spi_gd32_frame_exchange(dev);
 		if (ret < 0) {
 			break;
 		}
 	} while (spi_gd32_transfer_ongoing(data));
 
+#ifdef CONFIG_SPI_ASYNC
+	spi_context_complete(&data->ctx, ret);
```

Comment:
This is required for SPI_ASYNC, even in a polling case?

Reply:
In my understanding, SPI_ASYNC is just only notify finish of transfer with signal object.
Potentially it should works with polling implementations.
(I think it is enough useful for running with worker thread.)

For example, this PR work with enabling SPI_ASYNC even if not enabling SPI_GD32_INTERRUPT.

Reply:
Thanks for the explanation.